src/index.ts

```diff
@@ -35,6 +36,10 @@ export interface OptimizedPeak {
   shape: Shape1D;
 }
 
+type PeakWithIDOrNot<T extends Peak> = T extends { id: string }
```

Comment:
Why do you extend `Peak` and not `OptimizedPeak`?

Also why call it `PeakWithIDOrNot` rather than `OptimizedPeakWithIDOrNot` ?

Reply:
There T is the type of input data, that is why I'm using `Peak`.